device/samples/dmpatterns_fwupdate_device.js

```diff
@@ -7,13 +7,7 @@ var Protocol = require('azure-iot-device-mqtt').Mqtt;
 var url = require('url');
 var async = require('async');
 
-// receive the IoT Hub device connection string as a command line parameter
-if(process.argv.length < 3) {
-  console.error('Usage: node dmpatterns_fwupdate_device.js <<IoT Hub Device Connection String>>');
-  process.exit(1);
-}
-
-var connectionString = process.argv[2];
+var connectionString = process.env.IOTHUB_CONNECTION_STRING;
```

Comment:
>IOTHUB_CONNECTION_STRING [](start = 35, length = 24)

And how does going from command line argument to environment variable improve things.
Also see previous comment about the use of IOTHUB_CONNECTION_STRING.